src/index.js

```diff
@@ -266,24 +267,25 @@ class KadDHT {
 
           this._log('peers in rt: %d', rtp.length)
           if (rtp.length === 0) {
-            this._log.error('No peers from routing table!')
-            return cb(new Error('Failed to lookup key'))
+            const errMsg = 'Failed to lookup key! No peers from routing table!'
+
+            this._log.error(errMsg)
+            return cb(errcode(new Error(errMsg), 'ERR_NO_PEERS_IN_ROUTING_TABLE'))
           }
 
           // we have peers, lets do the actualy query to them
           const query = new Query(this, key, (peer, cb) => {
             this._getValueOrPeers(peer, key, (err, rec, peers) => {
               if (err) {
                 // If we have an invalid record we just want to continue and fetch a new one.
-                if (!(err instanceof errors.InvalidRecordError)) {
+                if (!(err.code === 'ERR_INVALID_RECORD')) {
                   return cb(err)
                 }
               }
 
               const res = { closerPeers: peers }
 
-              if ((rec && rec.value) ||
-                  err instanceof errors.InvalidRecordError) {
+              if ((rec && rec.value) || err.code === 'ERR_INVALID_RECORD') {
```

Comment:
Need to check the existence of err. If there is no record/record value and no error, this would break.
```suggestion
              if ((rec && rec.value) || (err && err.code === 'ERR_INVALID_RECORD')) {
```